client/packages/common/src/ui/forms/JsonForms/components/Array.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { ComponentType, useMemo } from 'react';
+import {
+  rankWith,
+  schemaTypeIs,
+  ArrayControlProps,
+  findUISchema,
+  JsonFormsUISchemaRegistryEntry,
+  ControlElement,
+  Layout,
+} from '@jsonforms/core';
+import {
+  withJsonFormsArrayControlProps,
+  JsonFormsDispatch,
+} from '@jsonforms/react';
+import {
+  Box,
+  Typography,
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+} from '@mui/material';
+import {
+  IconButton,
+  PlusCircleIcon,
+  MinusCircleIcon,
+  ChevronDownIcon,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { RegexUtils } from '@common/utils';
+import {
+  FORM_LABEL_COLUMN_WIDTH,
+  FORM_INPUT_COLUMN_WIDTH,
+} from '../styleConstants';
+import { JsonData } from '../useJsonForms';
+
+interface UISchemaWithCustomProps extends ControlElement {
+  defaultNewItem?: JsonData;
+  itemLabel?: string;
+}
+
+interface ArrayControlCustomProps extends ArrayControlProps {
+  uischema: UISchemaWithCustomProps;
+  removeItems: (path: string, toDelete: number[]) => () => void;
+  data: JsonData[];
+}
+
+interface FoundUiSchema extends Layout {
+  elements: ControlElement[];
+}
+
+export const arrayTester = rankWith(5, schemaTypeIs('array'));
+
+const UIComponent = (props: ArrayControlCustomProps) => {
+  const t = useTranslation('common');
+  const {
+    uischema,
+    uischemas,
+    schema,
+    path,
+    data,
+    addItem,
+    removeItems,
+    enabled,
+    label,
+    rootSchema,
+  } = props;
+
+  const foundUISchema = useMemo(
+    () =>
+      findUISchema(
+        uischemas as JsonFormsUISchemaRegistryEntry[],
+        schema,
+        uischema.scope,
+        path,
+        undefined,
+        uischema,
+        rootSchema
+      ),
+    []
+  ) as FoundUiSchema;
```

Comment:
This was the missing bit of magic I needed to get the uiSchema shape correct -- pasted from [here](https://github.com/eclipsesource/jsonforms/blob/413b9767ea8e2c0b6adb3919f5e6b76263d4be31/packages/material/src/layouts/ExpandPanelRenderer.tsx#L107-L119) at the suggestion of JsonForms developer.

Reply:
Empty dependency array as I don't see why this needs to be calculated more than once.

Reply:
you mean because all the params are not changing? it probably just for completeness or to make the linter happy (maybe useful to enable the linter? think its called exhaustive-deps)

Reply:
The empty dependency array just means it never gets recalculated. The [example I linked to](https://github.com/eclipsesource/jsonforms/blob/413b9767ea8e2c0b6adb3919f5e6b76263d4be31/packages/material/src/layouts/ExpandPanelRenderer.tsx#L107-L119) has a full dependency array, but they're all values that shouldn't change, so I think it makes it more explicit that `foundUISchema` is a fixed value if we leave the array empty.

It's not a big deal, was just giving an explanation for why my version looks slightly different to the one I copied it from.